discovery-provider/src/tasks/cache_user_balance.py

```diff
@@ -64,6 +80,8 @@ def get_immediate_refresh_user_ids(redis: Redis) -> List[int]:
 #        a) new (created_at == updated_at)
 #        b) not new, but stale: last updated prior to (now - threshold)
 #     we look up said users, adding User_Balance rows, and removing them from Redis.
+#     we check if they have associated_wallets and update those balances as well
```

Comment:
I know this wasn't from your PR, but the above comment lies now:
- we can take out the bit about if_current_user_exists on L73
- we don't do the check about new on line 80
- the slower balance refresh thing on L85 also isn't the case
- we should probably also mention something about the difference between the two sets